cupy/core/core.pyx

```diff
@@ -2118,6 +2126,62 @@ cdef _argmax = create_reduction_func(
     None, _min_max_preamble)
 
 
+cdef _round_preamble = '''
+__device__ long long pow10(long long n){
+  long long x = 1, a = 10;
+  while (n) {
+    if (n & 1) x *= a;
+    a *= a;
+    n >>= 1;
+  }
+  return x;
+}'''
+
+
+cdef _round_float = '''
+if (in1 == 0) {
+    out0 = round(in0);
+} else {
+    long long x = pow10(abs(in1));
```

Comment:
The range of `long long` seems too small to support the following.
```
>>> np.around(1.5e200, decimals=-200)
1.0000000000000003e+200
>>> np.around(1.5e200, decimals=-199)
1.5000000000000005e+200
>>> np.around(1.5e200, decimals=-201)
0.0
```

Reply:
Thanks, I fixed.